src/mscorlib/src/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -253,36 +253,34 @@ private unsafe int LastIndexOfCore(string source, string target, int startIndex,
             }
         }
 
-        private bool StartsWith(string source, string prefix, CompareOptions options)
+        private unsafe bool StartsWith(ReadOnlySpan<char> source, ReadOnlySpan<char> prefix, CompareOptions options)
         {
             Debug.Assert(!_invariantMode);
 
-            Debug.Assert(!string.IsNullOrEmpty(source));
-            Debug.Assert(!string.IsNullOrEmpty(prefix));
+            Debug.Assert(!source.IsEmpty);
+            Debug.Assert(!prefix.IsEmpty);
             Debug.Assert((options & (CompareOptions.Ordinal | CompareOptions.OrdinalIgnoreCase)) == 0);
 
-            if (_isAsciiEqualityOrdinal && CanUseAsciiOrdinalForOptions(options) && source.IsFastSort() && prefix.IsFastSort())
```

Comment:
What is the performance impact of the IsFastSort paths removal on the existing string methods?

Reply:
I will get a Linux VM, measure, and report back.

Reply:
The results on OSX seem way off (220 seconds vs 0.4 seconds) for Ascii string. I think we need the path that calls `IsPrefix(source, prefix, GetOrdinalCompareOptions(options));`

There is also a 5-10% regression on Windows for, possibly due to the overhead of the span constructor and Slice (smaller for value.Length > 100, larger for value.Length < 10).
2870 ms vs 3100 ms

Reply:
I changed StartsWith to just call IsPrefix (without the IsFastSort()/etc checks)

i.e.
```C#
private bool StartsWith(string source, string prefix, CompareOptions options)
{
    // Debug.Asserts
    return IsPrefix(source, prefix, GetOrdinalCompareOptions(options));
}
```

And it is still 3x slower on OSX: ~400 ms vs ~1300 ms

Reply:
cc @AndyAyersMS, can we do anything to reduce the overhead of string.AsReadOnlySpan() and MemoryMarshal.GetReference(span)?

https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/Span.NonGeneric.cs#L149
https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/ReadOnlySpan.Fast.cs#L115
https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/Runtime/InteropServices/MemoryMarshal.Fast.cs#L40

```C#
        public static ReadOnlySpan<char> AsReadOnlySpan(this string text)
        {
            if (text == null)
                return default;

            return new ReadOnlySpan<char>(ref text.GetRawStringData(), text.Length);
        }

        internal ref char GetRawStringData()
        {
            return ref _firstChar;
        }

        internal ReadOnlySpan(ref T ptr, int length)
        {
            Debug.Assert(length >= 0);

            _pointer = new ByReference<T>(ref ptr);
            _length = length;
        }

        public static ref T GetReference<T>(ReadOnlySpan<T> span) => ref span._pointer.Value;
```

Reply:
> I changed StartsWith to just call IsPrefix (without the IsFastSort()/etc. checks)
> And it is still 3x slower on OSX: ~400 ms vs ~1300 ms

To try to resolve this, I have to avoid the round-tripping that happens when we call IsPrefix (which calls span.StartsWith again).

If I create an internal span based StartsWithOrdinalHelper, we could call that method directly instead of calling IsPrefix (from CompareInfo.Unix.StartsWith).

That brings down the runtime to around ~800 ms.